File: clients/scala/src/main/scala/komus24/model/VehicleOrder.scala
package komus24.model

import komus24.util.StreamUtil

/**
 * Order for controlling a single vehicle
 *
 * @param accelerate Acceleration (-1 - fully backwards, +1 - fully forward)
 * @param brakes Hand brakes
 * @param rotate Rotate the steering wheel (-1 - full clockwise, +1 - full counterclockwise)
 */
case class VehicleOrder(accelerate: Double, brakes: Boolean, rotate: Double) {
    /**
     * Write VehicleOrder to output stream
     */
    def writeTo(stream: java.io.OutputStream): scala.Unit = {
        StreamUtil.writeDouble(stream, accelerate)
        StreamUtil.writeBoolean(stream, brakes)
        StreamUtil.writeDouble(stream, rotate)
    }

    /**
     * Get string representation of VehicleOrder
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("VehicleOrder { ")
        stringBuilder.append("accelerate: ")
        stringBuilder.append(accelerate)
        stringBuilder.append(", ")
        stringBuilder.append("brakes: ")
        stringBuilder.append(brakes)
        stringBuilder.append(", ")
        stringBuilder.append("rotate: ")
        stringBuilder.append(rotate)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object VehicleOrder {
    /**
     * Read VehicleOrder from input stream
     */
    def readFrom(stream: java.io.InputStream): VehicleOrder = VehicleOrder(
        StreamUtil.readDouble(stream),
        StreamUtil.readBoolean(stream),
        StreamUtil.readDouble(stream)
    )
}

File: clients/java/src/main/java/komus24/model/VehicleOrder.java
package komus24.model;

import komus24.util.StreamUtil;

/**
 * Order for controlling a single vehicle
 */
public class VehicleOrder {
    /**
     * Acceleration (-1 - fully backwards, +1 - fully forward)
     */
    private double accelerate;

    /**
     * Acceleration (-1 - fully backwards, +1 - fully forward)
     */
    public double getAccelerate() {
        return accelerate;
    }

    /**
     * Acceleration (-1 - fully backwards, +1 - fully forward)
     */
    public void setAccelerate(double value) {
        this.accelerate = value;
    }
    /**
     * Hand brakes
     */
    private boolean brakes;

    /**
     * Hand brakes
     */
    public boolean isBrakes() {
        return brakes;
    }

    /**
     * Hand brakes
     */
    public void setBrakes(boolean value) {
        this.brakes = value;
    }
    /**
     * Rotate the steering wheel (-1 - full clockwise, +1 - full counterclockwise)
     */
    private double rotate;

    /**
     * Rotate the steering wheel (-1 - full clockwise, +1 - full counterclockwise)
     */
    public double getRotate() {
        return rotate;
    }

    /**
     * Rotate the steering wheel (-1 - full clockwise, +1 - full counterclockwise)
     */
    public void setRotate(double value) {
        this.rotate = value;
    }

    public VehicleOrder(double accelerate, boolean brakes, double rotate) {
        this.accelerate = accelerate;
        this.brakes = brakes;
        this.rotate = rotate;
    }

    /**
     * Read VehicleOrder from input stream
     */
    public static VehicleOrder readFrom(java.io.InputStream stream) throws java.io.IOException {
        double accelerate;
        accelerate = StreamUtil.readDouble(stream);
        boolean brakes;
        brakes = StreamUtil.readBoolean(stream);
        double rotate;
        rotate = StreamUtil.readDouble(stream);
        return new VehicleOrder(accelerate, brakes, rotate);
    }

    /**
     * Write VehicleOrder to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeDouble(stream, accelerate);
        StreamUtil.writeBoolean(stream, brakes);
        StreamUtil.writeDouble(stream, rotate);
    }

    /**
     * Get string representation of VehicleOrder
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("VehicleOrder { ");
        stringBuilder.append("accelerate: ");
        stringBuilder.append(String.valueOf(accelerate));
        stringBuilder.append(", ");
        stringBuilder.append("brakes: ");
        stringBuilder.append(String.valueOf(brakes));
        stringBuilder.append(", ");
        stringBuilder.append("rotate: ");
        stringBuilder.append(String.valueOf(rotate));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}